### PR TITLE
Recog german (DateTime-Extractors and Bugfixes)

### PR DESCRIPTION
**Pull request by flo-zuewi**
Continued implementation of https://github.com/Microsoft/Recognizers-Text/issues/240

### CODE REVIEW

File: .NET/Microsoft.Recognizers.Text.DateTime/German/Extractors/GermanDateExtractorConfiguration.cs [hunk 1]
```diff
@@ -102,6 +102,12 @@ public class GermanDateExtractorConfiguration : BaseOptionsConfiguration,IDateEx
             // 23/7
             new Regex(DateTimeDefinitions.DateExtractor9, RegexOptions.IgnoreCase | RegexOptions.Singleline),
 
+            //10
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/German/Extractors/GermanDateExtractorConfiguration.cs]: Can you add examples in the code comment here that illustrate which cases would match this regex?

File: .NET/Microsoft.Recognizers.Text.DateTime/German/Extractors/GermanDateExtractorConfiguration.cs [hunk 2]
```diff
@@ -102,6 +102,12 @@ public class GermanDateExtractorConfiguration : BaseOptionsConfiguration,IDateEx
...
+            new Regex(DateTimeDefinitions.DateExtractor10, RegexOptions.IgnoreCase | RegexOptions.Singleline),
+
+            //11
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/German/Extractors/GermanDateExtractorConfiguration.cs]: Same

File: Specs/DateTime/German/MergedExtractor.json [hunk 3]
```diff
@@ -1 +1,144 @@
...
+    "NotSupported":"javascript",
+    "Results": [
+      {
+        "Text": "4 Uhr",
+        "Type": "time"
+      }
+    ]
+  }
+  ,
+  {//should be datetime
```
Comment (tellarin) [inline Specs/DateTime/German/MergedExtractor.json]: What do you mean? It should be recognized as datetime, but due to a bug falls here?

File: Specs/DateTime/German/MergedExtractor.json [hunk 4]
```diff
@@ -1 +1,144 @@
...
+    "NotSupportedByDesign": "python",
+    "NotSupported":"javascript",
+    "Results": [
+      {
+        "Text": "letzte Woche",
+        "Type": "daterange"
+      }
+    ]
+  },
+  {//should be datetimerange
```
Comment (tellarin) [inline Specs/DateTime/German/MergedExtractor.json]: Same question.

File: Specs/DateTime/German/MergedExtractor.json [hunk 5]
```diff
@@ -1 +1,144 @@
...
+    ]
+  },
+  {
+    "Input": "Das nächste meeting wird am 16. März 2017 sein. Wie wäre es denn mit einer Diskussionsrunde heute um 2 Uhr Nachmittags?",
+    "NotSupported": "javascript",
+    "NotSupportedByDesign": "python",
+    "Results": [
+      {
+        "Text": "16. März 2017 ",
+        "Type": "set" //should be date
```
Comment (tellarin) [inline Specs/DateTime/German/MergedExtractor.json]: If this is wrong, this test should not be enabled. Should be moved to the correct spec and marked as notsupported for now.

File: Specs/NumberWithUnit/German/AgeModel.json [hunk 6]
```diff
@@ -142,23 +142,6 @@
   ]
 }
 ,
-{//ordinals are currently not supported in the AgeModel
```
Comment (tellarin) [inline Specs/NumberWithUnit/German/AgeModel.json]: Why remove the test and not keep here as unsupported for now?

File: .NET/Microsoft.Recognizers.Text.DateTime/German/Extractors/GermanDateExtractorConfiguration.cs (hunk 1 above)
Comment (flo-zuewi) [reply .NET/Microsoft.Recognizers.Text.DateTime/German/Extractors/GermanDateExtractorConfiguration.cs]: Yes, I will do this with the next PR. I am still not fully done with the extractors, some of them will probably be deleted and others changed. DateExtractor 10 and 11 will most certainly be blended into the one or two of the other ones and removed after that.

File: Specs/DateTime/German/MergedExtractor.json (hunk 3 above)
Comment (flo-zuewi) [reply Specs/DateTime/German/MergedExtractor.json]: The set-extractor seems to be a little bit to "dominant",  looking further into this issue today, forgot to put the testcases back to not supported

File: Specs/NumberWithUnit/German/AgeModel.json (hunk 6 above)
Comment (tellarin) [reply Specs/NumberWithUnit/German/AgeModel.json]: OK

**Pull request merged**